enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,483 @@
+> 3. If the files are valid, and the index image tag exists, take a gitdiff between the present content of the index (i.e the existing json representations) and the content of the folder given as input. Display the diff as an output of the command with the following message 
+> "The following changes will be updated in the index:"
+> If there is no diff between the content in the container image and the content of the local folder, throw an error with the message "No diff found between index and local content to update."
+
+##### Introducing a new `create` subcommand for `opm index`
+
+A new subcommand `create` will be introduced under`opm index`, that will take a folder containing json representation of packages, and a container image tag as input, and will create a new index with the package representations fed to it. 
+Since `opm index add` will be modified to create the representation, the `create` subcommand will only serve as a UX facilitator for users. i.e underneath the hood, the `create` subcommand only needs to iterate over the pacakge representations and call the `opm index add` command with the appropriate `path` and `channel` info. 
+
+>Note: Best effort will be made to adhere to the UX mentioned above however appropriate fot this command too. 
+
+##### Introducing a new `validate` subcommand for `opm index`
+
+A new subcommand `validate` will be introduced under `opm index`, that will validate the content of each package's json representation. These validations will be in the context of rules about the structure of an index (required fields, name conventions etc of bundles, packages, and the index itself). A best effort is currently being made to aggregate these rules from various sources and maintained in the [api project's validation library](https://github.com/operator-framework/api/tree/master/pkg/validation). The `opm index validate` will be reponsible for validating the index for authors to make sure that the index conforms to the rules mentioned above. 
+
+#### Package cofig CUE definition
+
+CUE(https://cuelang.org/docs/about/) is a configuration and data validation language that makes it easy to define the structure of a resource and validate it's content. This section will discuss CUE as a possible implementation to define the structure of packages inside an index, and use that definition to generate the json representation of the packages, and ultimately the representation of the index. 
+
+##### Schema Defintion
+
+An example definition of a package that lives in the [api](https://github.com/operator-framework/api) can be: 
+
+```bash=
+$ cat package.cue
+#gvk: {
+    group: string
+    version: string
+    kind: string
+}
+
+#bundle: {
+    path: string
+    version: string
+    channel: string
+    upgradeMode: "semver"|"semver-skippatch"
```

Comment:
this isn't going to support replaces mode?

Reply:
@bparees [it will](https://github.com/operator-framework/enhancements/pull/37/files#diff-1edbfba9fdddd2b74b5cd32c6de146a4f62e1400a93bee78aed2cbf6ebaaa1d7R355)